sale_commission_advance/models/account_move.py

```diff
@@ -0,0 +1,129 @@
+        return any(
+            self.invoice_line_ids.mapped(
+                lambda x: x.sale_line_ids and x.sale_line_ids.is_downpayment
+            )
+        )
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.depends("move_id.partner_id")
+    def _compute_agent_ids(self):
+        self.agent_ids = False  # for resetting previous agents
+        down_payment_items = self.filtered(
+            lambda x: x.move_id.partner_id
+            and x.move_id.move_type[:3] == "out"
+            and x.sale_line_ids.is_downpayment
+        )
+        for record in down_payment_items:
+            agent_ids = record._prepare_agents_vals_down_payment()
+            record.update({"agent_ids": agent_ids})
+        regular_items = self.filtered(
+            lambda x: x.move_id.partner_id
+            and x.move_id.move_type[:3] == "out"
+            and not x.sale_line_ids.is_downpayment
+        )
+        if regular_items:
+            super(AccountMoveLine, regular_items)._compute_agent_ids()
+
+    def _prepare_agents_vals_partner(self, partner):
+        if not self.move_id.has_related_sale_with_down_payment():
+            return super()._prepare_agents_vals_partner(partner)
+        sol_agents = self.sale_line_ids.mapped("agent_ids").filtered(
+            lambda x: x.amount > 0
+        )
+        agent_ids = sol_agents.mapped("agent_id")
+        res = []
+        for agent_id in agent_ids:
+            res.append(
+                (
+                    0,
+                    0,
+                    {
+                        "agent_id": agent_id.id,
+                        "amount": 0,
+                        "commission_id": agent_id.commission_id.id,
+                    },
+                )
+            )
+        return res
+
+    def _prepare_agents_vals_down_payment(self):
+        # Add agent lines with amount > 0 and with total commission amount
+        self.sale_line_ids.ensure_one()
+        sale_order_id = self.sale_line_ids.order_id
+        agent_sol_ids = sale_order_id.order_line.filtered(
+            lambda x: x.is_downpayment is False and x.product_id and x.agent_ids
+        )
+        sol_agents = agent_sol_ids.mapped("agent_ids").filtered(lambda x: x.amount > 0)
+        agent_ids = sol_agents.mapped("agent_id")
+        res = []
+        for agent_id in agent_ids:
+            total = sum(
+                sol_agents.filtered(lambda x: x.agent_id == agent_id).mapped("amount")
+            )
+            res.append(
+                (
+                    0,
+                    0,
+                    {
+                        "agent_id": agent_id.id,
+                        "amount": 0,
+                        "downpayment_base_amount": total,
+                    },
+                )
+            )
+        return res
+
+
+class AccountInvoiceLineAgent(models.Model):
+    _inherit = "account.invoice.line.agent"
+
+    commission_id = fields.Many2one(
```

Comment:
Why redefining the field?

Reply:
to make it required=False

Reply:
Then just put such value, not the whole definition. Anyway, it's weird to have a commission line... without commission!

Reply:
Amount calculation is not based on commission_id in case of down payment. That's why it's ok to have commission line without commission but with amount. But anyway i've removed this from code. Now commission_id will be there, but amount is calculated as percent of down payment.